File: app/src/main/java/pozzo/apps/travelweather/ui/adapter/ForecastInfoWindowAdapter.java
package pozzo.apps.travelweather.ui.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.TextView;

import com.google.android.gms.maps.GoogleMap;
import com.google.android.gms.maps.model.Marker;

import pozzo.apps.travelweather.R;

/**
 * Create every single InfoWindow to show on maps.
 *
 * Created by sarge on 11/2/15.
 */
public class ForecastInfoWindowAdapter implements GoogleMap.InfoWindowAdapter {
	private Context context;
	private LayoutInflater inflater;

	public ForecastInfoWindowAdapter(Context context) {
		this.context = context;
		this.inflater = LayoutInflater.from(context);
	}

	@Override
	public View getInfoWindow(Marker marker) {
		return null;
	}

	@Override
	public View getInfoContents(Marker marker) {
		View contentView = inflater.inflate(R.layout.adapter_forecast, null);
		TextView lTitle = (TextView) contentView.findViewById(R.id.lTitle);
		lTitle.setText(marker.getTitle());
		return contentView;
	}
}

File: app/src/main/kotlin/pozzo/apps/travelweather/forecast/adapter/ForecastInfoWindowAdapter.kt
package pozzo.apps.travelweather.forecast.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import com.google.android.gms.maps.GoogleMap
import com.google.android.gms.maps.model.Marker
import pozzo.apps.travelweather.R
import pozzo.apps.travelweather.forecast.model.point.WeatherPoint

class ForecastInfoWindowAdapter(context: Context) : GoogleMap.InfoWindowAdapter {
    private val inflater: LayoutInflater = LayoutInflater.from(context)

    override fun getInfoWindow(marker: Marker): View? {
        return null
    }

    override fun getInfoContents(marker: Marker): View {
        val contentView = inflater.inflate(R.layout.adapter_forecast, null)
        val lTitle = contentView.findViewById<TextView>(R.id.lTitle)
        lTitle.text = marker.title

        val mapPoint = marker.tag
        if (mapPoint is WeatherPoint) {
            val poweredByImage = contentView.findViewById<ImageView>(R.id.poweredByImage)
            poweredByImage.setImageResource(mapPoint.poweredBy.icon)
        }
        return contentView
    }
}
